.travis.yml

```diff
@@ -49,7 +49,7 @@ deploy:
       all_branches: true # todo: this should only deploy on master once pypi is fully setup
       tags: false
       # only deploy from one build
-      condition: "$TRAVIS_OS_NAME == linux"
+      condition: "[$TRAVIS_OS_NAME == linux] && [[ $(python --version) == *3.8* ]]"
```

Comment:
This potentially matches Python versions that don't start with 3.8, like 2.3.8

How about
```
[[ $(python --version | cut -d' ' -f2) =~ ^3.8 ]]
```
This ensure the version starts with 3.8

Reply:
Good catch. I have implemented your change. Longer term I am planning to make the semantic meaning of "run only once" more explicit in the code however that will be a much bigger diff and a diffrent MR.